components/SetUpStaking/StakingStepper.jsx

```diff
@@ -0,0 +1,165 @@
+import { truncateAddress, formatToEth, getAddressFromBytes32 } from 'common-util/functions';
+import ConnectTwitterModal from '../ConnectTwitter/Modal';
+import { useTotalBond, checkAndApproveToken, createAndStake } from './requests';
+
+const { Paragraph, Text } = Typography;
+
+const XDAI_FOR_GAS = 2;
+
+const ConnectTwitter = ({ account }) => {
+  if (account) {
+    return (
+      <Text type="secondary">
+        Your connected Twitter account:{' '}
+        <a href={`https://twitter.com/${account}`} target="_blank">@{account} ↗</a>
+      </Text>
+    )
+  }
+  return (
+    <>
+      <Paragraph type="secondary">
+        Link your Twitter account to your Contribute profile.
+      </Paragraph>
+      <ConnectTwitterModal/>
+    </>
+  )
+}
+
+const SetUpAndStake = ({ disabled, twitterId, onNextStep }) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [multisig, setMultisig] = useState(null);
+
+  const account = useSelector((state) => state?.setup?.account);
+
+  const { totalBond, isLoading: isTotalBondLoading } = useTotalBond()
+
+  const handleSetUpAndStake = async () => {
+    if (!account) return;
+    if (!twitterId) return;
+    if (!totalBond) return;
+
+    setIsLoading(true);
+
+    try {
+      await checkAndApproveToken({
+        account,
+        amountToApprove: totalBond,
+      })
+      
+      const result = await createAndStake({
+        socialId: twitterId,
+      })
+
+      if (result) {
+        const logs = result.logs;
+        const createdAndStakedEvent = logs[logs.length - 1];
+        const multisig = getAddressFromBytes32(createdAndStakedEvent.topics[3]);
+        // TODO: write multisig to ceramic
+        setMultisig(multisig)
+        onNextStep()
+      }
+
+    } catch (error) {
+      notifyError('Error: could not set up & stake');
+      console.error(error);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  if (multisig) {
+    return (
+      <Text type="secondary">
+        Your contribute service Safe address:{' '}
+        <a href={`https://gnosisscan.io/address/${multisig}`} target="_blank">
+          {truncateAddress(multisig)} ↗
+        </a>
+      </Text>
+    )
+  }
+
+  return (
+    <>
+      <Paragraph type="secondary">
+        Ensure you have:
+        <ul>
+          <li>
+            {isTotalBondLoading ? <Skeleton.Button size="small"/> : `${formatToEth(totalBond, 0, 0)}`}{' '}
+            OLAS on Base Chain for staking
+          </li>
+          <li>{XDAI_FOR_GAS} XDAI on Base Chain for gas</li>
+        </ul>
+      </Paragraph>
+      <Button type="primary" disabled={disabled} loading={isLoading} onClick={handleSetUpAndStake}>
+        Set up & stake
+      </Button>
+    </>
+  )
+}
+
+const TweetAndEarn = ({ disabled }) => (
+  <>
+    <Paragraph type="secondary">
+      Visit Leaderboard and participate in tweet campaigns to earn points.
+      If you earn enough points for the epoch, you might be eligible to earn staking rewards.
```

Comment:
"might" is quite passive, negative leaning
"could" is less passive, positive leaning
```suggestion
      If you earn enough points for the epoch, you could be eligible to earn staking rewards.
```